lib/preload/ndb/preload.js

```diff
@@ -7,60 +7,42 @@
-    try {
-      if (!require('worker_threads').isMainThread)
-        return;
-    } catch (e) {
-      // node 8 does not support workers
-    }
-    let scriptName = '';
-    try {
-      scriptName = require.resolve(process.argv[1]);
-    } catch (e) {
-      // preload can get scriptName iff node starts with script as first argument,
-      // we should be ready for exception in other cases, e.g., node -e '...'
-    }
-    const ppid = process.env.NDD_PPID;
-    process.env.NDD_PPID = process.pid;
-    if (!process.env.NDD_DATA)
-      process.env.NDD_DATA = process.pid + '_ndbId';
-    process.versions['ndb'] = '1.1.3';
-    const inspector = require('inspector');
-    inspector.open(0, undefined, false);
-    const info = {
-      cwd: process.cwd(),
-      argv: process.argv.concat(process.execArgv),
-      data: process.env.NDD_DATA,
-      ppid: ppid,
-      id: String(process.pid),
-      inspectorUrl: inspector.url(),
-      scriptName: scriptName
-    };
-    const {execFileSync} = require('child_process');
-    execFileSync(process.execPath, [__filename], {
-      env: {
-        NDD_IPC: process.env.NDD_IPC,
-        NDD_PUBLISH_DATA: JSON.stringify(info)
-      }
-    });
-  } else {
-    const net = require('net');
-    const TIMEOUT = 30000;
-    const socket = net.createConnection(process.env.NDD_IPC, () => {
-      socket.write(process.env.NDD_PUBLISH_DATA);
-      const timeoutId = setTimeout(() => socket.destroy(), TIMEOUT);
-      socket.on('data', () => {
-        clearTimeout(timeoutId);
-        socket.destroy();
-      });
-    });
-    socket.on('error', err => {
-      process.stderr.write('\u001b[31mndb is not found:\u001b[0m\n');
-      process.stderr.write('please restart it and update env variables or unset NDD_IPC and NODE_OPTIONS.\n');
-      process.exit(0);
-    });
+  try {
+    if (!require('worker_threads').isMainThread)
+      return;
+  } catch (e) {
+    // node 8 does not support workers
   }
+  let scriptName = '';
+  try {
+    scriptName = require.resolve(process.argv[1]);
+  } catch (e) {
+    // preload can get scriptName iff node starts with script as first argument,
+    // we should be ready for exception in other cases, e.g., node -e '...'
+  }
+
+  const ppid = process.env.NDD_PPID;
+  process.env.NDD_PPID = process.pid;
+  if (!process.env.NDD_DATA)
+    process.env.NDD_DATA = process.pid + '_ndbId';
+  process.versions['ndb'] = '1.1.3';
+  const inspector = require('inspector');
+  inspector.open(0, undefined, false);
+  const base64 = new Buffer(JSON.stringify({
+    cwd: process.cwd(),
+    argv: process.argv.concat(process.execArgv),
+    data: process.env.NDD_DATA,
+    ppid: ppid,
+    id: String(process.pid),
+    inspectorUrl: inspector.url(),
+    scriptName: scriptName
+  })).toString('base64');
+  const { execFileSync } = require('child_process');
+  if (process.platform === 'win32')
+    execFileSync('cmd', ['/C', `echo "${base64}" > ${process.env.NDD_IPC}`]);
```

Comment:
We use execFileSync to:
- report data about process to ipc server,
- wait until frontend is ready, we emulate waiting for debugger here.